sample-azure-ha-yamls/azure-ha-kustomize-statefulset.yaml

```diff
@@ -0,0 +1,129 @@
+# Copyright Sonatype, Inc., 2023, All Rights Reserved.
+#
+# Use this statefulset.yaml when using Kustomize.
+#
+# Note: This YAML file is just an example and cannot be used as-is. 
+# You must fill it out with the appropriate information for your deployment to be able to use it.
+
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: nxrm-nexus-statefulset
+  namespace: nxrm
+  labels:
+    app: nxrm
+spec:
+  replicas: 2
+  serviceName: nxrm-nexus-statefulset-service
+  selector:
+    matchLabels:
+      app: nxrm
+  template:
+    metadata:
+      labels:
+        app: nxrm
+    spec:
+      initContainers:
+        # chown nexus-data to 'nexus' user and init log directories/files for a new pod
+        # otherwise the side car containers will crash a couple of times and backoff whilst waiting
+        # for nxrm-app to start and this increases the total start up time.
+        - name: chown-nexusdata-owner-to-nexus-and-init-log-dir
+          image: busybox:1.33.1
+          command: [/bin/sh]
+          args:
+            - -c
+            - >-
+              mkdir -p /nexus-data/etc/logback &&
+              mkdir -p /nexus-data/log/tasks &&
+              mkdir -p /nexus-data/log/audit &&
+              touch -a /nexus-data/log/tasks/allTasks.log &&
+              touch -a /nexus-data/log/audit/audit.log &&
+              touch -a /nexus-data/log/request.log &&
+              chown -R '200:200' /nexus-data
+          volumeMounts:
+            - name: nexusdata
+              mountPath: /nexus-data
+      terminationGracePeriodSeconds: 20
+      containers:
+      - name: nxrm-app
+        image: sonatype/nexus3:3.50.0
+        securityContext:
+          runAsUser: 200
+        imagePullPolicy: IfNotPresent
+        ports:
+        - containerPort: 8081
+        env:
+        - name: DB_NAME
+          value: <db-name>
+       -  name: DB_PORT
+          value: 3306
```

Comment:
Is this port right? It isn't the typical postgresql port

Reply:
Set to 5432